BrainUp/BrainUp/Auth/AuthVM.swift

```diff
@@ -0,0 +1,54 @@
+//
+//  AuthVM.swift
+//  BrainUp
+//
+//  Created by Evgenii Zhigunov on 3/9/22.
+//
+
+import Foundation
+import FirebaseCore
+import FirebaseAuth
+
+protocol AuthVMProtocol: AnyObject {
+    func auth(login: String, password: String)
+    func register(login: String, password: String)
+}
+
+class AuthVM: AuthVMProtocol {
+    private weak var view: BasicViewInterface?
+    private var authStateHandle: AuthStateDidChangeListenerHandle!
+    
+    init(view: BasicViewInterface) {
+        self.view = view
+        authStateHandle = Auth.auth().addStateDidChangeListener { _, user in
```

Comment:
It would be better for testing to create some protocol that contains func like addStateDidChangeListener, signIn etc. Subscribe Auth to this protocol and inject it though init, function or even property. 

Example:
var auth: AuthProvider = Auth.auth()

extension Auth : AuthProvider {}